File: ccc07s4.py
points = int(input())

#Slides from a point
slides = [[] for x in range(points+1)]

#Number of paths to the slide
paths = [0 for x in range(points+1)]
#First point
paths[1] = 1

while True:
	slide = input().split()
	if slide == ['0', '0']:
		break
	slides[int(slide[0])].append(int(slide[1]))

for point in range(1, points+1):
	for dest in slides[point]:
		paths[dest] += paths[point]


print(paths[-1])

File: ccc07s4.cpp
#include <iostream>
#include <vector>
#include <array>
#include <string>

using namespace std;

int main() {
	int n;
	cin >> n;

	int paths[9999] = {1};
	paths[1] = 1;

	vector<int> connections[n+1];

	int x, y;

	while (true) {
		cin >> x;
		cin >> y;
		if (x + y == 0) {
			break;
		}
		connections[x].push_back(y);
	}

	for (int i = 1; i != n+1; i++) {
		for (int slide : connections[i]) {
			paths[slide] += paths[i];	

		}

	}

	cout << paths[n] << endl;
	



}

